src/sinks/splunk_hec/logs/sink.rs

```diff
@@ -34,6 +33,7 @@ pub struct HecLogsSink<S> {
     pub indexed_fields: Vec<String>,
     pub host: String,
     pub timestamp_nanos_key: Option<String>,
+    pub timestamp_key: String,
```

Comment:
I haven't followed this, or looked at the original issue (should add that in the description so it will auto close) - but did we consider an option that takes a boolean?

Reply:
(updated the description to reference the original issue)

Reply:
Ah, yeah, I had suggested the behavior implemented here since it seemed consistent with our other `_key` fields on components. See the issue for more details: https://github.com/vectordotdev/vector/issues/12128

Reply:
👍 that's fair - I don't love our usage of "empty string" means don't set the key, but we have been doing so.